src/main/java/com/snowflake/kafka/connector/internal/streaming/DirectTopicPartitionChannel.java

```diff
@@ -80,6 +80,12 @@ public class DirectTopicPartitionChannel implements TopicPartitionChannel {
   private final AtomicLong processedOffset =
       new AtomicLong(NO_OFFSET_TOKEN_REGISTERED_IN_SNOWFLAKE);
 
+  // This offset is would not be required for buffer-less channel, but we add it to keep buffered
+  // and non-buffered
+  // channel versions compatible.
+  private final AtomicLong latestConsumerOffset =
```

Comment:
Nit: since this offset represents a current consumer group offset let's name it `currentConsumerGroupOffset`. I was always confused by `latestConsumerOffset`